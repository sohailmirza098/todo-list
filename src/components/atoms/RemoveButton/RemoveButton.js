// import React from 'react'

// export default function RemoveButton(props) {
//     const btnStyle = {
//         background: '#ff0000',
//         color: '#fff',
//         border: 'none',
//         padding: '5px 9px',
//         borderRadius: '50%',
//         cursor: 'pointer',
//         float: 'right'
//     }
//     return (
//         <button onClick={() => props.delTodo(props.id)} style={btnStyle}>x</button>
//     )
// }


import React from 'react'

const btnStyle = {
    background: '#ff0000',
    color: '#fff',
    border: 'none',
    padding: '5px 9px',
    borderRadius: '50%',
    cursor: 'pointer',
    float: 'right'
}

const RemoveButton = (props) => {
    console.log(props);
    return(<button onClick={() => {
        props.delTodo(props.id)
    }} style={btnStyle}>{props.value}</button>);
}

export default RemoveButton; 