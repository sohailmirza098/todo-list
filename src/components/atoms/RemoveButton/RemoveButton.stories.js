import React from 'react';
import RemoveButton from "./RemoveButton";
import { action } from '@storybook/addon-actions';
import { withKnobs, text } from '@storybook/addon-knobs/react';

export default {
  title: 'RemoveButton',
  component: RemoveButton,
  // decorators: [withKnobs]
};

const buttonData = {
  hidden: 'false',
  value: 'x',
  delTodo: action("Button clicked"),
}

export const RemoveButtonStory = () => {

    return(<RemoveButton {...buttonData}></RemoveButton>);
};
