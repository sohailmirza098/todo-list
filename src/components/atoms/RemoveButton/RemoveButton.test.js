import React from "react";
import { create } from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import RemoveButton from "./RemoveButton";

describe("Remove Button testing", () => {

  test("Matches the snapshot", () => {
    const input = create(<RemoveButton />);
    expect(input.toJSON()).toMatchSnapshot();
  });

  test("RemoveButton is present", () => {
    const {getByRole}=  render(<RemoveButton />);
    const button = getByRole("button");
    expect(button).toBeInTheDocument();
  });

  test("RemoveButton has content X", () => {
    const {getByRole}=  render(<RemoveButton />);
    const button = getByRole("button");
    expect(button).toHaveTextContent("x");
  });

  test("RemoveButton onclick",()=>{
    const mockClickMethod = jest.fn();
    const {getByRole}=  render(<RemoveButton onClick={mockClickMethod()}/>);
    fireEvent.click(getByRole("button"));
    expect(mockClickMethod).toHaveBeenCalledTimes(1);
  });
});