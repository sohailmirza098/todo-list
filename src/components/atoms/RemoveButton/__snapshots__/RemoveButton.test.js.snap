exports[`Remove Button testing Matches the snapshot 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "background": "#ff0000",
      "border": "none",
      "borderRadius": "50%",
      "color": "#fff",
      "cursor": "pointer",
      "float": "right",
      "padding": "5px 9px",
    }
  }>
  x
</button>
`;
